contribute/architecture/backend/communication.md

```diff
@@ -2,134 +2,107 @@
 
 Grafana uses a _bus_ to pass messages between different parts of the application. All communication over the bus happens synchronously.
 
-> **Deprecated:** The bus has officially been deprecated, however, we're still using the command/query objects paradigms.
+## Commands and queries
 
-There are three types of messages: _events_, _commands_, and _queries_.
+Grafana structures arguments to [services](services.md) using a command/query
+separation where commands are instructions for a mutation and queries retrieve
+records from a service.
 
-## Events
+Services should define their methods as `func[T, U any](ctx context.Context, args T) (U, error)`
+or `func[T, U any](ctx context.Context, args T) U` for infallible methods
```

Comment:
> infallible methods

I don't understand this. 

In general, an example method might help

Reply:
I've changed the language to be more Go-centric.

I think I can squeeze in an example method here before merge, since we had that before and it could be considered a regression that we don't do now.